jupyter_dashboards/nbextension/notebook/dashboard-view/layout/grid/layout.js

```diff
@@ -600,24 +468,29 @@ define([
         this.gridstack.remove_widget($cell, false /* don't detach node */);
     };
 
-    // move cell from hidden table to main grid
-    Dashboard.prototype._showCell = function($cell, constraints, row) {
-        // compute correct dimensions (taking into account any given `constraints`)
-        var dim = this._computeCellDimensions($cell, constraints);
+    // computes metadata for a cell that doesn't have grid layout metadata
+    GridLayout.prototype._addNotebookCellToDashboard = function($cell) {
+        var dim = this._computeCellDimensions($cell);
+        return dim.isEmpty ? Promise.resolve() : this._showCell($cell);
```

Comment:
Should that promise resolve with `$cell` in order to match promise returned from `_showCell`?